manifests/params.pp

```diff
@@ -19,8 +19,17 @@
     $datadir    = '/etc/puppetlabs/puppet/hieradata'
     $owner      = 'pe-puppet'
     $group      = 'pe-puppet'
-    $provider   = 'pe_gem'
     $confdir    = '/etc/puppetlabs/puppet'
+    if $::puppetversion =~ /3.7/ {
+      $provider = 'pe_puppetserver_gem'
+      # 3.7 also needs /opt/puppet/bin/eyaml
+      exec { 'install pe_gem':
```

Comment:
Unfortunately that would only work on puppet 4.0.0 and later; https://tickets.puppetlabs.com/browse/PUP-1073